contracts/libs/data-structures/IncrementalMerkleTree.sol

```diff
@@ -108,6 +140,14 @@ library IncrementalMerkleTree {
         IMT _tree;
     }
 
+    /**
+     * @notice The Bytes32 Incremental Merkle Tree constructor, creates a tree instance with the
+     * given size, O(1) complex
+     */
+    function newBytes32(uint256 size_) internal pure returns (Bytes32IMT memory imt) {
```

Comment:
If these functions are constructors, then they should always be called (otherwise the tree is in the wrong state). Should we add a constructor without size_? And it is not size_ TBH, it is height